src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Lower.java

```diff
@@ -3896,11 +3884,364 @@ public void visitForLoop(JCForLoop tree) {
     public void visitReturn(JCReturn tree) {
         if (tree.expr != null)
             tree.expr = translate(tree.expr,
-                                  types.erasure(currentMethodDef
-                                                .restype.type));
+                                  currentRestype);
         result = tree;
     }
 
+    @Override
+    public void visitLambda(JCLambda tree) {
+        Type expectedRet = types.findDescriptorType(tree.type).getReturnType();
+        int prevVariableIndex = variableIndex;
+        try {
+            variableIndex = 0;
+            if (tree.body.hasTag(JCTree.Tag.BLOCK)) {
+                Type prevRestype = currentRestype;
+                try {
+                    currentRestype = expectedRet;
+                    super.visitLambda(tree);
+                } finally {
+                    currentRestype = prevRestype;
+                }
+            } else {
+                tree.body = translate((JCExpression) tree.body, expectedRet);
+            }
+            result = tree;
+        } finally {
+            variableIndex = prevVariableIndex;
+        }
+    }
+
+    @Override
+    public void visitReference(JCMemberReference tree) {
+        if (needsConversionToLambda(tree)) {
+            // Convert to a lambda, and process as such
+            MemberReferenceToLambda conv = new MemberReferenceToLambda(tree);
+            result = translate(conv.lambda());
+        } else {
+            super.visitReference(tree);
+        }
+    }
+    // where
+        boolean needsVarArgsConversion(JCMemberReference tree) {
+            return tree.varargsElement != null;
+        }
+
+        /**
+         * @return Is this an array operation like clone()
+         */
+        boolean isArrayOp(JCMemberReference tree) {
+            return tree.sym.owner == syms.arrayClass;
+        }
+
+        boolean receiverAccessible(JCMemberReference tree) {
+            //hack needed to workaround 292 bug (7087658)
```

Comment:
I agree there's a lot of cruft in here. We started adding predicates simply because the 292 machinery could not support all invocation modes. But then we also added other stuff: all the accessibility checks are there because if javac decides to generate an accessibility bridge for the method, then we can't use a MH directly (as the target method would be inaccessible - the bridge should be called instead). In these cases, it was preferred to just let `Lower` do its job, and translate the method call correctly, using bridges etc.

But, with respect to this specific issue, I agree, `ownerAccessible` is probably no longer needed, as the underlying MH deficiency was addressed long ago.